src/ex1/script.js

```diff
@@ -0,0 +1,129 @@
+const sortButton = document.querySelector(".name-sort");
+
+// Event trigger when user clicks the add task button
+addTaskButton.addEventListener("click", () => {
+   addTask();
+});
+
+// Same as the button add task but only when user press "Enter"
+taskInput.addEventListener("keypress", (e) => {
+   if (e.key === "Enter") {
+      addTask();
+   }
+});
+
+// Clears all tasks when clear all button is clicked
+clearButton.onclick = () => {
+   document
+      .querySelectorAll(".task-container")
+      .forEach((task) => task.remove());
+   pendingTasksUpdate(0);
+};
+
+// Sorts the tasks containers by name
+sortButton.onclick = () => {
+   const taskContainers = [...document.querySelectorAll(".task-container")];
+   taskContainers.sort((a, b) =>
+      a.children[0].innerText.localeCompare(b.children[0].innerText)
+   );
+   taskContainers.map((container) => container.remove());
+   taskContainers.map((container) => allTasksContainer.appendChild(container));
+};
+
+// Adds the user task input to the tasks list
+function addTask() {
+   // Checks for blank spaces
+   if (taskInput.value.trim().length === 0) {
+      alert("Please fill in a task");
+      taskInput.value = "";
+   } else {
+      const taskContainer = createTaskContainer();
+      const newTask = createNewTask(taskContainer);
+      const deleteTask = createDeleteTaskButton(taskContainer);
+      addHoverReveal(taskContainer, deleteTask, newTask);
+      createTaskAnimation(newTask);
+      taskInput.value = "";
+      pendingTasksUpdate("+");
+   }
+}
+
+// Creates container which holds task box and delete button
+function createTaskContainer() {
+   const taskContainer = document.createElement("div");
+   taskContainer.classList.add("task-container");
+   allTasksContainer.append(taskContainer);
+
+   return taskContainer;
+}
+
+// Creates task box and click alert event to it
+function createNewTask(taskContainer) {
+   const newTask = document.createElement("div");
+   newTask.textContent = taskInput.value;
+   newTask.classList.add("task");
+   taskContainer.append(newTask);
+
+   newTask.addEventListener("click", () => {
+      alert(newTask.textContent);
+   });
+
+   return newTask;
+}
+
+// Creates the delete button next to each task
+function createDeleteTaskButton(taskContainer) {
+   const deleteTask = document.createElement("div");
+   const i = document.createElement("i");
+   i.classList.add("fa-solid", "fa-trash");
+   deleteTask.appendChild(i);
+   deleteTask.classList.add("delete");
+   taskContainer.append(deleteTask);
+
+   deleteTask.addEventListener("click", () => {
+      taskContainer.remove();
+      pendingTasksUpdate("-");
+   });
+
+   return deleteTask;
+}
+
+// Adds mouseover events to reveal the delete button only when user hovers
+function addHoverReveal(taskContainer, deleteTask, newTask) {
+   taskContainer.addEventListener("mouseover", () => {
+      deleteTask.style.display = "flex";
+      newTask.style.width = "26rem";
```

Comment:
it's better to apply a class with these properties, and name the class as what it represents, for example `deleted-btn-mouseover`, `new-task-mouseover`, rather than explicitly set the properties on the html elements one by one using javascript (hard to maintain)

Reply:
Oh sounds like best practice. Corrected and added the classes to the new constants.js file